### PR TITLE
Fix WKT parser

### PR DESCRIPTION
**Pull request by lbartoletti**
## Description

This PR will fix WKT parser.  This should fix these errors:

```g = QgsGeometry.fromWkt('MULTILINESTRING((((1 2, 2 4, 4 5))')```
returns 
```<QgsGeometry: MultiLineString ((0 2, 2 4, 4 5))>```

and 
```
QgsGeometry.fromWkt('Point ((5, 5))')
```
returns
```
<QgsGeometry: Point (0 0)>
```
instead of null (or empty?) geometry

from #34496

Follow-up #36578

### CODE REVIEW

File: tests/src/python/test_qgsgeometry.py [hunk 1]
```diff
@@ -842,7 +842,7 @@ def testReferenceGeometry(self):
                                        "Perimeter {}: mismatch Expected:\n{}\nGot:\n{}\n".format(i + 1, exp, result))
 
     def testCollection(self):
-        g = QgsGeometry.fromWkt('MultiLineString()')
```
Comment (nyalldawson) [inline tests/src/python/test_qgsgeometry.py:845]: I can't see a test for `MultiLineString()` in the newly added TestQgsGeometry::wktParser() test -- can you add one please?

File: tests/src/python/test_qgsgeometry.py [hunk 2]
```diff
@@ -4244,42 +4244,42 @@ def testGeometryDisplayString(self):
 
     def testDeleteVertexCircularString(self):
 
-        wkt = "CircularString ((0 0,1 1,2 0))"
```
Comment (nyalldawson) [inline tests/src/python/test_qgsgeometry.py:4247]: Can you also add this one to TestQgsGeometry::wktParser()

File: tests/testdata/relates_data.csv [hunk 3]
```diff
@@ -47,7 +47,7 @@ LINESTRING(20 20,100 20,20 20)|LINESTRING(60 20,200 20)|101FFF102
 LINESTRING(40 60,180 60,180 140,100 140,100 60,220 60,220 180,80 180,80 60,280 60)|LINESTRING(140 60,180 60,220 60,260 60)|101FF0FF2
 POINT(20 20)|POLYGON((60 120,60 40,160 40,160 120,60 120))|FF0FFF212
 MULTIPOINT(0 20,40 20)|POLYGON((20 40,20 0,60 0,60 40,20 40))|0F0FFF212
-MULTIPOINT(0 20,20 20))|POLYGON((20 40,20 0,60 0,60 40,20 40))|F00FFF212
```
Comment (nyalldawson) [inline tests/testdata/relates_data.csv:50]: This one too

File: tests/src/python/test_qgsgeometry.py (hunk 1 above)
Comment (lbartoletti) [reply tests/src/python/test_qgsgeometry.py:845]: Done.

File: tests/src/python/test_qgsgeometry.py (hunk 2 above)
Comment (lbartoletti) [reply tests/src/python/test_qgsgeometry.py:4247]: Done

File: tests/testdata/relates_data.csv (hunk 3 above)
Comment (lbartoletti) [reply tests/testdata/relates_data.csv:50]: Done

File: src/core/geometry/qgsgeometryutils.cpp [hunk 4]
```diff
@@ -1109,6 +1110,12 @@ QgsPointSequence QgsGeometryUtils::pointsFromWKT( const QString &wktCoordinateLi
   for ( const QString &pointCoordinates : coordList )
   {
     QStringList coordinates = pointCoordinates.split( rx, QString::SkipEmptyParts );
+
+    // exit with an empty set if one list contains invalid value.
+    QRegularExpression rxIsNumber( QStringLiteral( "^[+-]?(\\d\\.?\\d*[Ee][+\\-]?\\d+|(\\d+\\.\\d*|\\d*\\.\\d+)|\\d+)$" ) );
```
[the remark this answers is not in the document]
Comment (lbartoletti) [reply src/core/geometry/qgsgeometryutils.cpp]: done. Thanks

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
Thanks @lbartoletti !

**Comment by lbartoletti** [comment]
> Thanks @lbartoletti !

And thanks to you for the review!

**Pull request merged**